packages/deploy-config-generator-shared/src/base/generator.ts

```diff
@@ -0,0 +1,49 @@
+import Generator from 'yeoman-generator';
+import { t } from '../utils';
+import { DefaultLogger, LogWrapper, type ILogWrapper } from '@sap-ux/fiori-generator-shared';
+import type { LogLevel, IVSCodeExtLogger } from '@vscode-logging/logger';
+
+/**
+ * Base class for deployment generators.
+ */
+export abstract class DeploymentGenerator extends Generator {
+    private static _logger: ILogWrapper = DefaultLogger;
+
+    /**
+     * Deployment generator constructor.
+     *
+     * @param args - arguments passed to the generator
+     * @param options - options passed to the generator
+     */
+    constructor(args: string | string[], options: Generator.GeneratorOptions) {
```

Comment:
Just a query. Do we need to define the options supported when launching the deployment config generator?